File: Serenity.Script.Core/Q/Q.ErrorHandling.cs
using System;
using System.Runtime.CompilerServices;

namespace Serenity
{
    public static partial class Q
    {
        public static class ErrorHandling
        {
            public static void ShowServiceError(ServiceError error)
            {
                if (error == null)
                    throw new Exception("error is null!");

                Q.Alert(error == null ? "??ERROR??" : (error.Message ?? error.Code));
            }
        }

        public static void TryCatch(this Action<object> fail, Action callback)
        {
            if (fail == null)
            {
                callback();
                return;
            }

            try
            {
                callback();
            }
            catch (Exception ex)
            {
                fail(ex);
            }
        }
    }
}

File: Serenity.TypeScript.CoreLib/Q/Q.ErrorHandling.ts

namespace Q {

    export namespace ErrorHandling {

        export function showServiceError(error: Serenity.ServiceError) {
            let msg: any;
            if (error == null) {
                msg = '??ERROR??';
            }
            else {
                msg = error.Message;
                if (msg == null) {
                    msg = error.Code;
                }
            }

            Q.alert(msg);
        }
    }
}